senders/proxy_test.go

```diff
@@ -22,26 +21,67 @@ func netcat(addr string, keepopen bool) {
 	lis.Close()
 }
 
-func init() {
+func TestProxySends(t *testing.T) {
 	go netcat("localhost:30000", false)
 	go netcat("localhost:40000", false)
 	go netcat("localhost:50000", false)
+
+	time.Sleep(time.Second)
```

Comment:
Consider adding a TODO here to see if there is a better way to avoid the race condition. I asked on the golang group, but waiting for a response.

Reply:
Add TODO